arangod/IResearch/IResearchFeature.cpp

```diff
@@ -407,10 +411,231 @@ void registerFilters(arangodb::aql::AqlFunctionFeature& functions) {
   addFunction(functions, { "ANALYZER", ".,.", flags, &contextFunc });  // (filter expression, analyzer)
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief IResearchLinkCoordinator-specific implementation of an
+///        IndexTypeFactory
+////////////////////////////////////////////////////////////////////////////////
+struct IResearchLinkCoordinatorFactory : public arangodb::IndexTypeFactory {
```

Comment:
Let`s keep factories within corresponding links (as nested, but now public classes). This will allow us to keep  link constructors private and keep link instantination process encapsulated. Also this will make code support simpler as related entities will be stored together.

Reply:
@Dronplane Would you consider this sufficiently addressed by the latest changes?